ci/run_python_tests.sh

```diff
@@ -0,0 +1,183 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# Functions
+function fn_die() {
+  echo -e "$1" >&2
+  exit "${2:-1}"
+}
+
+function import_gpg_keys() {
+  # shellcheck disable=SC2207
+  declare -r my_arr=( $(echo "${@}" | tr " " "\n") )
+
+  if [ "${#my_arr[@]}" -eq 0 ]; then
+    fn_die "Error: There are ZERO gpg keys to import. ZEN_REPO_MAINTAINER_KEYS variable is not set. Exiting ..."
+  else
+    # shellcheck disable=SC2145
+    printf "%s\n" "Tagged build, fetching keys:" "${@}" ""
+    for key in "${my_arr[@]}"; do
+      gpg -v --batch --keyserver hkps://keys.openpgp.org --recv-keys "${key}" ||
+      gpg -v --batch --keyserver hkp://keyserver.ubuntu.com --recv-keys "${key}" ||
+      gpg -v --batch --keyserver hkp://pgp.mit.edu:80 --recv-keys "${key}" ||
+      gpg -v --batch --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys "${key}" ||
+      { echo -e "Warning: ${key} can not be found on GPG key servers. Please upload it to at least one of the following GPG key servers:\nhttps://keys.openpgp.org/\nhttps://keyserver.ubuntu.com/\nhttps://pgp.mit.edu/"; export IS_A_RELEASE="false"; }
+    done
+  fi
+}
+
+function check_signed_tag() {
+  local tag="${1}"
+
+  if git verify-tag -v "${tag}"; then
+    echo "${tag} is a valid signed tag"
+  else
+    fn_die "Error: ${tag} signature is NOT valid. Exiting ..."
```

Comment:
I think you do not want to exit here just to give a message and set IS_A_RELASE or the variable you are using to FALSE? Similar what is going on under import_keys funtion

```
else
    echo "" && echo "=== Warning: GIT's tag = ${tag} signature is NOT valid. The build is not going to be released ... ===" && echo ""
    export IS_A_RELEASE="false"
  fi
```